src/test/scala/org/ergoplatform/nodeView/state/ErgoStateContextSpec.scala

```diff
@@ -58,13 +58,27 @@ class ErgoStateContextSpec extends HistoryTestHelpers {
       fb.copy(extension = newExtension)
     }
 
+    // regression stress-test for sporadic failure https://github.com/ergoplatform/ergo/issues/2114
+    // disabling for now
+    // for (i <- 1 to 1000) {
+    // printf("\r%d05 ", i)
+
     // checks, specific for extension
     // validation of field keys size
     val imvKey = extensionKvGen(Extension.FieldKeySize - 1, Extension.FieldValueMaxSize).sample.get
     sc.appendFullBlock(fbWithFields(imvKey +: oldFields)) shouldBe 'failure
 
+    // https://github.com/ergoplatform/ergo/issues/2114
+    // this fails sporadically, when `(imvValue._1.head == 0)`, because less value bytes will be generated
+    // by extensionKvGen(). Workaround is to just generate again while `(imvValue._1.head == 0)`
+    // TODO: document, and possibly rewrite extensionKvGen after the above issues are clarified/solved
```

Comment:
likely 4 bytes output was made for a reason, so better to use another Array[Byte] generator here

Reply:
> likely 4 bytes output was made for a reason

...which is undocumented.

> better to use another Array[Byte] generator here

already mentioned within:

- https://github.com/ergoplatform/ergo/issues/2118

"Possibly introduce extensionKvGenValidated()"

(work on https://github.com/ergoplatform/ergo/issues/2118 should show what is the prefered way to go, documentation needs to be added there anyways)

Reply:
changed "possibly rewrite" to "possibly rewrite/replace"